### ISSUE TITLE
Boolean custom field icon lacks a text alternative

### ISSUE THREAD

**Issue by patrickhlauke** [bitwarden/mobile#1987]
### Steps To Reproduce

1. Use a screen reader (VoiceOver on iOS/Talkback on Android)
2. Go to an entry with a custom boolean field
3. Navigate to the custom boolean field

### Expected Result

A screen reader show announce the boolean as either checked/unchecked, on/off, true/false, or similar

### Actual Result

The custom character/icon is passed to screen readers as the only value. As this is a custom character, it's not announced at all - VoiceOver / Talkback are silent when reaching the character.

### Screenshots or Videos

iOS/VoiceOver screenshot - once focus is on the custom checked checkbox icon, note the voice output (bottom of the screen) is just showing the unrecognised custom character. Nothing is announced.

![iOS screenshot](https://user-images.githubusercontent.com/895831/178105830-d65b2a07-2bf5-4719-96b1-8274475242fc.png)

---

Android/Talkback screenshot - once focus is on the custom checked checkbox icon, note the voice output (bottom of the screen) is just showing the unrecognised custom character. Nothing is announced. 

![image0](https://user-images.githubusercontent.com/895831/178105929-64c9f618-ba13-4d83-a7e1-b7ad86a3a6b0.jpg)

### Additional Context

Note that in the extension, and in the desktop app, the boolean checkbox is announced as `true` or `false`.

Example using desktop app and NVDA/Windows
![desktop app and NVDA speech output, showing it announces the unchecked checkbox as "false"](https://user-images.githubusercontent.com/895831/178105974-b46de3fc-cf53-4f00-8ada-3122ed6ea249.png)

Example using chrome extension and NVDA/Windows
![chrome extension and NVDA speech output, showing it announces the unchecked checkbox as "false"](https://user-images.githubusercontent.com/895831/178105992-b5526fbf-1b35-4ce1-976a-f07f2a318e91.png)

This is done with a visually-hidden, but exposed to AT, `<span>` element that contains the actual value of the field

![desktop app with developer tools open, showing the hidden span after the checkbox icon containing the value of the field](https://user-images.githubusercontent.com/895831/178106024-efb383ff-e404-4dc4-9579-e953316cc8b8.png)

### Operating System

Android, iOS

### Operating System Version

Latest

### Device

iPhone SE (2020), Motorola moto g8

### Build Version

2022.6.0 (1859) iOS / 2022.6.1 (4795) Android

### Beta

- [ ] Using a pre-release version of the application.

**Comment by AE542** [comment]
Hi there, I'm thinking that just adding an accessibility hint to this field would allow it to read back if the box is checked or unchecked in VoiceOver. I'd like to try and tackle this, but I'd like to ask, where exactly is swift file with the custom field icon? 

Thanks for your time.

**Comment by cwickes** [comment]
From what I can tell, this is no longer a problem on iOS. Can we close this issue?

**Comment by patrickhlauke** [comment]
> From what I can tell, this is no longer a problem on iOS. Can we close this issue?

still the exact same problem on iOS with latest version (2023.3.2 (3195))

![Photo 30-04-2023, 19 58 43](https://user-images.githubusercontent.com/895831/235371546-f5a9561b-62ec-4f90-be7f-25d5cbad0da7.png)

**Comment by TomerPacific** [comment]
@patrickhlauke - Do you need assistance with this issue?

**Comment by patrickhlauke** [comment]
well if somebody wants to submit a fix (essentially taking the same approach as the website/extension), sure. have at it. i've scaled back my contributions here a bit since one things seemed to slow to a crawl at one point in terms of reviewing and merging...

**Comment by TomerPacific** [comment]
@patrickhlauke  - Thanks for your reply. Who is the person I should talk to about this?

**Comment by djsmith85** [comment]
@patrickhlauke I'm really sorry for your poor experience with that specific PR you are referring to. Trying to learn from these things to create a better experience for any contributor.

@TomerPacific: As for guidance on getting started, please have a look at Bitwarden's [Contributing Guidelines](https://contributing.bitwarden.com/). This should give you everything you need to get your environment set-up and further guidance around contributing in general.

Kind regards,
Daniel

**Comment by patrickhlauke** [comment]
@djsmith85 sorry, was clearly in a bit of a mood last night when i commented above. but yes more constructively, a bit more communication on some of the hanging PRs (where possible/appropriate) would help with just letting external contributors know what's happening.

**Comment by TomerPacific** [comment]
@djsmith85 - Reading the Contributing Guidelines and having a Windows machine, it seems like it would take some time for me to set up a local development environment properly. Wanted to communicate this and to allow anyone else to work on this issue if they want to.

**Comment by patrickhlauke** [comment]
so, as bitwarden/mobile has now been archived, can this issue be reopened here? @djsmith85 @vvolkgang

**Comment by djsmith85** [comment]
@patrickhlauke Thanks for the ping. 

I spoke with @vvolkgang and as this was reported for the old mobile client, could you please open a new ticket in case this is still an issue with the new mobile clients. Apologies for the back and forth with the shift of repos.

Kind regards,
Daniel
